File: apps/api/internal/contents/domain/content.go
package domain

import (
	"encoding/json"
	"time"
)

type ContentStatus string

const (
	ContentStatusDraft     ContentStatus = "draft"
	ContentStatusPublished ContentStatus = "published"
	ContentStatusArchived  ContentStatus = "archived"
)

func (s ContentStatus) IsValid() bool {
	switch s {
	case ContentStatusDraft, ContentStatusPublished, ContentStatusArchived:
		return true
	default:
		return false
	}
}

type ContentVisibility string

const (
	ContentVisibilityPrivate ContentVisibility = "private"
	ContentVisibilityPublic  ContentVisibility = "public"
)

func (v ContentVisibility) IsValid() bool {
	switch v {
	case ContentVisibilityPrivate, ContentVisibilityPublic:
		return true
	default:
		return false
	}
}

type Content struct {
	ID          string
	UserID      string
	CategoryID  *string
	Title       string
	Slug        *string
	Summary     *string
	Content     json.RawMessage
	Status      ContentStatus
	Visibility  ContentVisibility
	IsFavorite  bool
	CreatedAt   time.Time
	UpdatedAt   time.Time
	PublishedAt *time.Time
	DeletedAt   *time.Time
	DeleteAfter *time.Time
}

File: apps/web/src/features/contents/types/content.ts
export type ContentStatus = "draft" | "published" | "archived"
export type ContentVisibility = "private" | "public"

export interface Content {
  id: string
  user_id: string
  category_id: string | null
  title: string
  slug: string | null
  summary: string | null
  content: unknown
  status: ContentStatus
  visibility: ContentVisibility
  is_favorite: boolean
  created_at: string
  updated_at: string
  published_at: string | null
  deleted_at: string | null
  delete_after: string | null
}

export interface CreateContentPayload {
  title: string
  summary?: string | null
  content: unknown
  category_id?: string | null
  status?: ContentStatus
  visibility?: ContentVisibility
  is_favorite?: boolean
}

export interface UpdateContentPayload {
  title?: string
  summary?: string | null
  content?: unknown
  category_id?: string | null
  is_favorite?: boolean
}
